### PR TITLE
[IOTDB-643] Concurrent queries cause BufferUnderflowException when storage in HDFS

### PR DESCRIPTION
**Pull request by samperson1997**
Fix concurrent queries cause BufferUnderflowException when storage in HDFS
* Change methods in HDFSInput to synchronized
* Add lock to TsFileSequenceReader.readHeadMagic

### CODE REVIEW

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java [hunk 1]
```diff
@@ -225,10 +226,13 @@ public String readHeadMagic() throws IOException {
    * to the end of the magic head string.
    */
   public String readHeadMagic(boolean movePosition) throws IOException {
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java:228]: this parameter is no need ,  the tsfileInput is positioned in selfcheck already.

File: hadoop/src/main/java/org/apache/iotdb/hadoop/fileSystem/HDFSInput.java [hunk 2]
```diff
@@ -68,26 +60,26 @@ public long position() throws IOException {
   }
 
   @Override
-  public TsFileInput position(long newPosition) throws IOException {
+  public synchronized TsFileInput position(long newPosition) throws IOException {
```
Comment (qiaojialin) [inline hadoop/src/main/java/org/apache/iotdb/hadoop/fileSystem/HDFSInput.java:63]: this function should be used cautiously, it's better to use the read(position, length)

File: server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkMetadataCache.java [hunk 3]
```diff
@@ -100,18 +100,24 @@ public static ChunkMetadataCache getInstance() {
...
       throws IOException {
     if (!cacheEnable) {
-      // bloom filter part
-      TsFileSequenceReader tsFileReader = FileReaderManager.getInstance().get(filePath, true);
-      BloomFilter bloomFilter = tsFileReader.readBloomFilter();
-      if (bloomFilter != null && !bloomFilter.contains(seriesPath.getFullPath())) {
-        if (logger.isDebugEnabled()) {
-          logger.debug(String
-              .format("path not found by bloom filter, file is: %s, path is: %s", filePath, seriesPath));
+      lock.writeLock().lock();
```
[the remark this answers is not in the document]
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/engine/cache/ChunkMetadataCache.java:103]: Deleted

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java (hunk 1 above)
Comment (samperson1997) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java:228]: Removed

File: hadoop/src/main/java/org/apache/iotdb/hadoop/fileSystem/HDFSInput.java (hunk 2 above)
Comment (samperson1997) [reply hadoop/src/main/java/org/apache/iotdb/hadoop/fileSystem/HDFSInput.java:63]: Thanks for your reminding.

**Pull request merged**